frontend/packages/dev-console/src/components/topology2/layouts/TopologyColaLayout.ts

```diff
@@ -0,0 +1,97 @@
+import { ColaLayout, ColaNode, ColaGroup, ColaLink } from '@console/topology';
+import { TYPE_EVENT_SOURCE_LINK, TYPE_KNATIVE_SERVICE, TYPE_REVISION_TRAFFIC } from '../const';
+
+const getRevisionTraffic = (revision: ColaNode, edges: ColaLink[]): number => {
+  const edge = edges.find(
+    (e) =>
+      e.element.getType() === TYPE_REVISION_TRAFFIC && e.element.getTarget() === revision.element,
+  );
+  return edge ? edge.element.getData().data.percent : 0;
+};
+
+export default class TopologyColaLayout extends ColaLayout {
+  protected getConstraints(nodes: ColaNode[], groups: ColaGroup[], edges: ColaLink[]): any[] {
+    const constraints: any[] = [];
+    const revisionSeparation = 100;
+    const eventSourceLinkSeparation = 85;
+    const eventSourceLinkDist = 200;
+
+    if (groups) {
+      groups.forEach((g) => {
+        // Line up revisions of a knative service in a horizontal line
+        if (g.element.getType() === TYPE_KNATIVE_SERVICE) {
+          const serviceConstraint: any = {
+            type: 'alignment',
+            axis: 'y',
+            offsets: [],
+          };
+
+          // Sort revisions such that most traffic is to the right
+          const revisions = g.leaves.sort((n1, n2) => {
+            const traffic1 = getRevisionTraffic(n1, edges);
+            const traffic2 = getRevisionTraffic(n1, edges);
+            const val = traffic1 - traffic2;
+            if (val === 0) {
+              return n1.element.getLabel().localeCompare(n2.element.getLabel());
```

Comment:
You be comparing the data and not the label.

Reply:
My mistake. The above is when traffic data is equal. Comparing labels is to ensure more consistent sort order when traffic is equal.